# ECON 497 Midterm 1

## Q3

In [5]:
data <- read.csv("data/WAGE2.csv")

In [6]:
head(data,3)

wage hours IQ  KWW educ exper tenure age married black south urban sibs
1 769  40     93 35  12   11     2     31  1       0     0     1     1   
2 808  50    119 41  18   11    16     37  1       0     0     1     1   
3 825  40    108 46  14   11     9     33  1       0     0     1     1   
  brthord meduc feduc
1    2     8     8   
2 -999    14    14   
3    2    14    14

### Q3a

In [10]:
lm1 <- lm(log(wage) ~ age + educ + I(educ^2)
          , data=data)

In [11]:
summary(lm1)


Call:
lm(formula = log(wage) ~ age + educ + I(educ^2), data = data)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.85221 -0.23148  0.02332  0.26599  1.35680 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)  4.074297   0.584760   6.967 6.10e-12 ***
age          0.024134   0.004227   5.710 1.52e-08 ***
educ         0.220060   0.078347   2.809  0.00508 ** 
I(educ^2)   -0.005679   0.002776  -2.046  0.04106 *  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.3937 on 931 degrees of freedom
Multiple R-squared:  0.1288,	Adjusted R-squared:  0.126 
F-statistic: 45.87 on 3 and 931 DF,  p-value: < 2.2e-16


#### The coefficient $\beta_2$ gives marginal (multiplicative) effect of increasing *age*, when controlled for *educ*.

The R output above gives the value of the $t$ statistic $t := \frac{\beta_2 - 0}{se(\beta_2)} = 5.710$.
The statistic itself is $t_{n-1}$ distributed under Gauss-Markov assumptions. The rejection region for a 2-sided test at $5\% = 2\times2.5\%$ significance is { |t| > critical value calculated below }

In [12]:
qt(0.975,df=930)

[1] 1.962518

In this case, the value of the t-statistic is much greater than the critical value, we *reject* the null hypothesis that $\beta_2=0$.

### Q3c

as in the calculation below, the expected wage is expected to increase by approx. 52%

In [25]:
exp(lm1$coefficients["educ"]  * (12-10) + lm1$coefficients["I(educ^2)"] * (12-10)^2)

educ 
1.518016

### Q3d

We use the packaged version of the Breusch-Pagan test, and outline its functionality in words. In my submission of assignment 2, I have reproduced the BP test in code.

We take the null hypothesis of homoskedasticity, i.e. that the error terms $\epsilon_i$ are independent of any factors, in particular, they are not dependent on the factors of the original regression. Since we do not observe the error terms but only their realizations in the form of *residuals*, we therefore test for goodness-of-fit of the (studentized) auxiliary regression
$$\frac{e^2}{\hat{\sigma}^2} = \gamma_0 + \gamma_1 (age) + \gamma_2 (educ) + \gamma_3 (educ^2) + \eta$$
with $\hat{\sigma}^2 = \sum e_i^2/n$
We can apply either an F-version of the test; the standard model adequacy test on the auxiliary regression, which should reject model adequacy to confirm homoskedasticity.
Alternatively, we note that $\frac{SSR_{BP}}{n}\sim\chi^2_3$, and reject for larger-than-critical values of the $\chi^2$-statistic.

In [37]:
library(lmtest)

In [27]:
bptest(lm1)


	studentized Breusch-Pagan test

data:  lm1
BP = 2.0113, df = 3, p-value = 0.5701


At any reasonable significance level, e.g. $\alpha<10\%$, we *fail to reject* the null hypothesis of homoskedasticity.

In [28]:
data$pe <- data$age - data$educ - 6

In [31]:
head(data[,c("age","educ","pe")],3)

age educ pe
1 31  12   13
2 37  18   13
3 33  14   13

in the model log(wage) ~ age + educ + pe, the last column is an *exact linear combination of the first two*. This leads to unrealiable estimators of the regression coefficients, as they are inflated by $(1-R^2_k)^{-1}$, which in this case will be unbounded.

In [33]:
lm1.mult <- lm(log(wage) ~ age + educ + pe
              , data = data)

In [35]:
library(car)

In [36]:
car::vif(lm1.mult)

ERROR: Error in vif.default(lm1.mult): there are aliased coefficients in the model


## Misc

### Calculations for the theory questions

In [3]:
((1-0.23)/0.23)*(92/3)

[1] 102.6667

In [4]:
qf(0.95,df1=3,df2=92)

[1] 2.703594